### PR TITLE
[cglm] New port

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?
  Adds new port for [cglm](https://github.com/recp/cglm)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Note sure yet, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/cglm/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "cglm",
+  "version-semver": "0.8.8",
```
Comment (chausner) [inline ports/cglm/vcpkg.json:3]: Adheres to semver according to https://github.com/recp/cglm/blob/master/docs/source/version.rst.

**Review by Adela0814 — DISMISSED** [review]

**Review by BillyONeal — COMMENTED** [review]
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.  
  
Can you please confirm that the following looks correct? (The upstream instructions use `add_subdirectory` :(
```
cglm provides CMake targets:

    find_package(cglm CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cglm::cglm)
```
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

**Comment by chausner** [comment]
> Can you please confirm that the following looks correct? (The upstream instructions use add_subdirectory :(

Yes that should be correct but I have not tested it. They do write in their README (albeit without an example):

> or you can use find_package to configure cglm

[In their tests](https://github.com/recp/cglm/blob/7f1c062696eafb2bdc381315157288db2a8acfae/test/CMakeLists.txt#L29), they link with

```cmake
target_link_libraries(${TEST_MAIN} PRIVATE cglm)
```
but I assume this is equivalent to linking with `cglm::cglm`.

**Review by BillyONeal — APPROVED** [review]
Added usage. Will merge as soon as build comes back. Thank you!

**Pull request merged**